### PR TITLE
More simplification

### PR DESCRIPTION
**Pull request by sigmaSd**

### CODE REVIEW

File: rustio/src/client.rs [hunk 1]
```diff
@@ -82,15 +82,11 @@ impl Client {
         Self::send_get_request(url).unwrap().json().unwrap()
     }
 
-    pub fn get_station_by_id(&self, id: i32) -> Result<Station,&str> {
+    pub fn get_station_by_id(&self, id: i32) -> Option<Station> {
```
Comment (haecker-felix) [inline rustio/src/client.rs]: Why the change from Result to Option? Many errors can happen here (I/O error, network error, wrong ID,...). So it definitely make sense to use Result here.

But I have to admit, correct error handling is here still missing (see readme.md).

File: rustio/src/client.rs [hunk 2]
```diff
@@ -82,15 +82,11 @@ impl Client {
...
         let url = format!("{}{}{}", BASE_URL, STATION_BY_ID, id);
         let mut result : Vec<Station> = Self::send_get_request(url).unwrap().json().unwrap();
 
-        if result.len() > 0 {
-            Ok(result.remove(0))
-        }else {
-            Err("ID points to an empty station")
-        }
+        result.pop()
```
Comment (haecker-felix) [inline rustio/src/client.rs]: *Normally* there should be always a station available to return. And if not, it should be definitely correctly handled by the caller, because this behavior is not indented/wanted.

**Comment by sigmaSd** [comment]
Make sense I'll stop playing around with this  workaround, This should be handled correctly once and for all , I'll close this for now.
